### PR TITLE
fix(avaxp): use "P-" prefix in avaxp addresses

### PR DESCRIPTION
**Pull request by CapnMigraine**
Remove methods convert avaxp addresses by adding or removing the `P-` prefix.

Ticket: STLX-18215

### CODE REVIEW

File: modules/sdk-coin-avaxp/src/lib/utils.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class Utils implements BaseUtils {
    */
   /** @inheritdoc */
   isValidAddress(address: string): boolean {
-    return (address as string).match(/^[a-zA-Z0-9]+$/) !== null;
+    return /^P-[a-zA-Z0-9]+$/.test(address);
```
Comment (jicli) [inline modules/sdk-coin-avaxp/src/lib/utils.ts:40]: https://github.com/BitGo/bitgo-microservices/pull/21198#discussion_r927758537

Comment (jicli) [reply modules/sdk-coin-avaxp/src/lib/utils.ts:40]: and also, it's not always start with `P-` cuz node uses it, can we do something like 
return `/^P-[a-zA-Z0-9]+$/.test(address); || /^NodeId-[a-zA-Z0-9]+$/.test(address)`

File: modules/sdk-coin-avaxp/test/unit/lib/utils.ts [hunk 2]
```diff
@@ -3,23 +3,43 @@ import { Utils, KeyPair } from '../../../src/lib/';
...
+    it('should fail to validate invalid address', function () {
+      const address = 'fuji15jamwukfqkwhe8z26tjqxejtjd3jk9vj4kmxwa';
+      Utils.isValidAddress(address).should.be.false();
+    });
+
+    it('should validate an address', function () {
+      const address = 'P-fuji15jamwukfqkwhe8z26tjqxejtjd3jk9vj4kmxwa';
+      Utils.isValidAddress(address).should.be.true();
+    });
+
```
Comment (jicli) [inline modules/sdk-coin-avaxp/test/unit/lib/utils.ts:34]: add a unit test for nodeid? thanks!

File: modules/sdk-coin-avaxp/src/lib/utils.ts (hunk 1 above)
Comment (CapnMigraine) [reply modules/sdk-coin-avaxp/src/lib/utils.ts]: fixed

File: modules/sdk-coin-avaxp/test/unit/lib/utils.ts (hunk 2 above)
Comment (CapnMigraine) [reply modules/sdk-coin-avaxp/test/unit/lib/utils.ts:40]: done

**Review by jicli — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**